nmt.go

```diff
@@ -142,35 +181,59 @@ func (n *NamespacedMerkleTree) ProveRange(start, end int) (Proof, error) {
 
 // ProveNamespace returns a range proof for the given NamespaceID.
 //
-// In case the underlying tree contains leaves with the given namespace
-// their start and end index will be returned together with a range proof and
-// the found leaves. In that case the returned leafHash will be nil.
+// case 1) If the namespace nID is out of the range of the tree's min and max namespace
+// i.e., (nID < n.minNID) or (n.maxNID < nID)
+// ProveNamespace returns an error and does not generate any range proof,
+// instead it returns an empty Proof with empty nodes and the range (0,0) i.e.,
+// Proof.start = 0 and Proof.end = 0
+// to indicate that this namespace is not contained in the tree.
```

Comment:
[optional] can all comments be wrapped to 80 characters for codebase consistency? If you use VSCode, you can use https://github.com/stkb/Rewrap to do this for you